renderdoc/driver/metal/metal_types.h

```diff
@@ -262,6 +267,43 @@ struct VertexDescriptor
   rdcarray<VertexAttributeDescriptor> attributes;
 };
 
+// MTLAttributeDescriptor : based on the interface defined in
+// Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLStageInputOutputDescriptor.h
+struct AttributeDescriptor
+{
+  AttributeDescriptor() = default;
+  AttributeDescriptor(MTL::AttributeDescriptor *objc);
+  void CopyTo(MTL::AttributeDescriptor *objc);
+  NS::UInteger bufferIndex = 0;
+  NS::UInteger offset = 0;
+  MTL::AttributeFormat format = MTL::AttributeFormatInvalid;
+};
+
+// MTLBufferLayoutDescriptor : based on the interface defined in
+// Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLStageInputOutputDescriptor.h
+struct BufferLayoutDescriptor
+{
+  BufferLayoutDescriptor() = default;
+  BufferLayoutDescriptor(MTL::BufferLayoutDescriptor *objc);
+  void CopyTo(MTL::BufferLayoutDescriptor *objc);
+  NS::UInteger stride = 0;
+  MTL::StepFunction stepFunction = MTL::StepFunctionConstant;
+  NS::UInteger stepRate = 0;
+};
+
+// MTLStageInputOutputDescriptor : based on the interface defined in
+// Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLStageInputOutputDescriptor.h
```

Comment:
I am interested in your system setup (I have only ever been able to compare/use Metal development on my own one system. For me the Metal files are in MacOSX14.2.sdk i.e.

Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLStageInputOutputDescriptor.h

but for you you have Metal files in MacOSX12.1.sdk and MacOSX14.2.sdk

I suspect the files are quite similar in both places. The reason I mention it I found it a bit tricky to keep track of any MaxOS changes to Metal header files. I now keep a local copy of the Metal header files at a certain OS version i.e. currently the project is on MacOSX 14.0 then when a new Xcode or MacOS is released with new Metal version. I copy the headers over into a new folder and manually compare the Metal headers to find any changes and hand copy the changes over into the wrapped classes.

I checked locally and for MTLStageInputOutputDescriptor.h MacOSX14.0 and MacOSX12.1 versions are basically identical.

Reply:
Oh this probably a typo, as I only have the 14.2 SDKs locally, I'll correct it.